src/base64.h

```diff
@@ -0,0 +1,205 @@
+#ifndef SRC_BASE64_H_
+#define SRC_BASE64_H_
+
+#include "util.h"
+
+#include <stddef.h>
+
+namespace node {
+//// Base 64 ////
+#define base64_encoded_size(size) ((size + 2 - ((size + 2) % 3)) / 3 * 4)
+
+
+// Doesn't check for padding at the end.  Can be 1-2 bytes over.
+static inline size_t base64_decoded_size_fast(size_t size) {
+  size_t remainder = size % 4;
+
+  size = (size / 4) * 3;
+  if (remainder) {
+    if (size == 0 && remainder == 1) {
+      // special case: 1-byte input cannot be decoded
+      size = 0;
+    } else {
+      // non-padded input, add 1 or 2 extra bytes
+      size += 1 + (remainder == 3);
+    }
+  }
+
+  return size;
+}
+
+template <typename TypeName>
+size_t base64_decoded_size(const TypeName* src, size_t size) {
+  if (size == 0)
+    return 0;
+
+  if (src[size - 1] == '=')
+    size--;
+  if (size > 0 && src[size - 1] == '=')
+    size--;
+
+  return base64_decoded_size_fast(size);
+}
+
+
+// supports regular and URL-safe base64
+static const int8_t unbase64_table[] =
```

Comment:
Can you leave the table itself in string_bytes.cc?  Just drop the `static` keyword and declare it here.  Maybe add `[256]` so there is no confusion about its size.